src/System.Security.Cryptography.Algorithms/src/Internal/Cryptography/RandomNumberGeneratorImplementation.cs

```diff
@@ -10,11 +10,11 @@ internal sealed partial class RandomNumberGeneratorImplementation : RandomNumber
     {
         // As long as each implementation can provide a static GetBytes(ref byte buf, int length)
         // they can share this one implementation of FillSpan.
-        internal static void FillSpan(Span<byte> data)
+        internal static unsafe void FillSpan(Span<byte> data)
         {
             if (data.Length > 0)
             {
-                GetBytes(ref MemoryMarshal.GetReference(data), data.Length);
+                fixed (byte* ptr = data) GetBytes(ptr, data.Length);
```

Comment:
@jkotas, probably wouldn't make any difference perf-wise given the expense of GetBytes, but would the resulting asm be better/smaller if this were instead `fixed (byte* ptr = &data[0])`?

Reply:
With the current state of JIT optimizations, the most significant micro-improvement would be changing `> 0`  to `! = 0`. Here are the code sizes of different variants:

```csharp
// Codesize: 0xb0
if (data.Length != 0)
{
    fixed (byte* ptr = data) GetBytes(ptr, data.Length);
}
```

```csharp
// Codesize: 0xbb
if (data.Length > 0)
{
    fixed (byte* ptr = data[0]) GetBytes(ptr, data.Length);
}
```

```csharp
// Codesize: 0xc5
if (data.Length > 0)
{
    fixed (byte* ptr = data[0]) GetBytes(ptr, data.Length);
}
```

Reply:
The small test that I have done had `GetBytes` as PInvoke. It is not actually the case here so the absolute numbers are not representative. The relative numbers should be still valid.

Reply:
>  the most significant micro-improvement would be changing > 0 to ! = 0

 is it worth? Do I have to do?

Reply:
No, thanks.